pkg/scheduler/api/compatibility/compatibility_test.go

```diff
@@ -1254,16 +1239,47 @@ func TestCompatibility_v1_Scheduler(t *testing.T) {
 			schedulerframework.NewDefaultRegistry(),
 			nil,
 			[]kubeschedulerconfig.PluginConfig{},
-		); err != nil {
+		)
+		if err != nil {
 			t.Errorf("%s: Error constructing: %v", v, err)
 			continue
 		}
-	}
-
-	if !seenPredicates.HasAll(registeredPredicates.List()...) {
```

Comment:
I feel like this could still be there (it doesn't depend on the policy).

Reply:
From what I understand, It's only testing the policy decoding as well since seenPredicates are extracted from the policy directly.

Reply:
I mean, the check can still be there, but using the set that is calculated from `sched.Algorithm`